pulp_ansible/app/galaxy/v3/views.py

```diff
@@ -155,32 +156,10 @@ def get_queryset(self):
             collection=OuterRef("pk"), repository_version=repo_version
         )
         collections = Collection.objects.filter(versions__in=repo_version.content).distinct()
-        collections = collections.annotate(deprecated=Exists(deprecated_query))
-
-        versions_qs = CollectionVersion.objects.filter(pk__in=repo_version.content).values_list(
-            "collection_id",
-            "namespace",
-            "name",
-            "version",
-            "pulp_created",
+        collections = collections.annotate(
+            deprecated=Exists(deprecated_query), available_versions=ArrayAgg("versions__version")
```

Comment:
@cutwater I think we can solve the semver comparison with this, we bring all available versions and sort the values at the serializer
```python
version = sorted(
            obj.available_versions, key=lambda ver: semantic_version.Version(ver), reverse=True
        )[0]
```